src/celeritas/user/detail/SDParams.cc

```diff
@@ -0,0 +1,83 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/user/detail/SDParams.cc
+//---------------------------------------------------------------------------//
+#include "SDParams.hh"
+
+#include "corecel/data/CollectionBuilder.hh"
+#include "corecel/io/Join.hh"
+#include "geocel/GeoVolumeFinder.hh"
+
+namespace celeritas
+{
+
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct from list of volume labels.
+ */
+SDParams::SDParams(VecLabel volume_labels, GeoParamsInterface const& geo)
```

Comment:
Since `volume_labels` is only used locally and never modified, the argument should be a const ref.